fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -188,6 +198,140 @@ public void processDropMaterializedView(DropMaterializedViewStmt stmt) throws Dd
+            if (newMvName != null) {
+                processRenameMaterializedView(oldMvName, newMvName, db, materializedView);
+            } else if (refreshSchemeDesc != null) {
+                //change refresh scheme
+                processChangeRefreshScheme(refreshSchemeDesc, materializedView);
+            } else {
+                throw new DdlException("unsupported modification for materialized view");
+            }
+        } finally {
+            db.writeUnlock();
+        }
+    }
+
+    private void processChangeRefreshScheme(RefreshSchemeDesc refreshSchemeDesc, MaterializedView materializedView)
+            throws DdlException {
+        final MaterializedView.MvRefreshScheme oldRefreshScheme = materializedView.getRefreshScheme();
+        final MaterializedView.MvRefreshScheme newMvRefreshScheme = new MaterializedView.MvRefreshScheme();
+        final String refreshType = refreshSchemeDesc.getType().name();
+
+        if (refreshSchemeDesc instanceof SyncRefreshSchemeDesc) {
+            throw new DdlException("unsupported change to SYNC refresh type");
+        }
+
+        if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) {
+            AsyncRefreshSchemeDesc asyncRefreshSchemeDesc = (AsyncRefreshSchemeDesc) refreshSchemeDesc;
+            newMvRefreshScheme.setAsyncRefreshContext(oldRefreshScheme.getAsyncRefreshContext());
+            final MaterializedView.AsyncRefreshContext asyncRefreshContext = new MaterializedView.AsyncRefreshContext();
+            asyncRefreshContext.setStartTime(Utils.getLongFromDateTime(asyncRefreshSchemeDesc.getStartTime()));
+            asyncRefreshContext.setStep(asyncRefreshSchemeDesc.getStep());
+            asyncRefreshContext.setTimeUnit(asyncRefreshSchemeDesc.getTimeUnit());
+            newMvRefreshScheme.setAsyncRefreshContext(asyncRefreshContext);
+        }
+
+        final RefreshType newRefreshType = RefreshType.valueOf(refreshType);
+        newMvRefreshScheme.setType(newRefreshType);
+        newMvRefreshScheme.setLastRefreshTime(oldRefreshScheme.getLastRefreshTime());
+        materializedView.setRefreshScheme(newMvRefreshScheme);
+        final ChangeMaterializedViewRefreshSchemeLog log =
+                new ChangeMaterializedViewRefreshSchemeLog(materializedView, newMvRefreshScheme);
+        EditLog editLog = GlobalStateMgr.getCurrentState().getEditLog();
+        editLog.logMvChangeRefreshScheme(log);
+    }
+
+    private void processRenameMaterializedView(String oldMvName, String newMvName, Database db,
+                                               MaterializedView materializedView) throws DdlException {
+        if (oldMvName.equals(newMvName)) {
+            throw new DdlException("Same materialized view name");
+        }
+        if (db.getTable(newMvName) != null) {
+            throw new DdlException("Materialized view [" + newMvName + "] is already used");
+        }
+        materializedView.setName(newMvName);
+        db.createTable(materializedView);
```

Comment:
should drop and recreate table like OlapTable
it is my fault.

Reply:
That's all right.